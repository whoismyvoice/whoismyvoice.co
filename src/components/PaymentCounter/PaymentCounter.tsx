import * as React from 'react';

// Style
import '../../styles/paymentCounter.scss';

interface Props {
  numContributors: number;
  payment: string;
}

class PaymentCounter extends React.Component<Props> {
  render(): JSX.Element {
    const numbers = this.props.payment.split('');
    const digits = numbers.map((result, idx) => {
      return (
        <span className="counter__digit" key={idx}>
          {result}
          <span className="counter__gradient-top" />
        </span>
      );
    });

    return (
      <span className="member__payment">
        <span className="counter__accept">Received</span>
        <span className="counter__digit digit__margin">
          $
          <span className="counter__gradient-top" />
        </span>
        {digits}
        <span className="counter__from">
          from their top {this.props.numContributors} contributing industries
        </span>
      </span>
    );
  }
}

export default PaymentCounter;
