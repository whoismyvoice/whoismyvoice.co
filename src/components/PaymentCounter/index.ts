import Component from './PaymentCounter';

export default Component;
