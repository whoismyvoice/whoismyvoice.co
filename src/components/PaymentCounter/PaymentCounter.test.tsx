import * as React from 'react';
import * as renderer from 'react-test-renderer';
import PaymentCounter from './PaymentCounter';

jest.mock('mixpanel-browser');

it('renders correctly payment=0', () => {
  const tree = renderer
    .create(<PaymentCounter numContributors={5} payment={0} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly payment=1000', () => {
  const tree = renderer
    .create(<PaymentCounter numContributors={5} payment={1000} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly payment=50000', () => {
  const tree = renderer
    .create(<PaymentCounter numContributors={10} payment={50000} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
