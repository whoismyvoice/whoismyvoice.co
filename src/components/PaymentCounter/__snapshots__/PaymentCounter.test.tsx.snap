// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly payment=0 1`] = `
<span
  className="member__payment"
>
  <span
    className="counter__accept"
  >
    Received
  </span>
  <span
    className="counter__digit digit__margin"
  >
    $
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__digit"
  >
    0
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__from"
  >
    from their top 
    5
     contributing industries
  </span>
</span>
`;

exports[`renders correctly payment=1000 1`] = `
<span
  className="member__payment"
>
  <span
    className="counter__accept"
  >
    Received
  </span>
  <span
    className="counter__digit digit__margin"
  >
    $
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__digit"
  >
    1
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__digit"
  >
    0
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__digit"
  >
    0
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__digit"
  >
    0
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__from"
  >
    from their top 
    5
     contributing industries
  </span>
</span>
`;

exports[`renders correctly payment=50000 1`] = `
<span
  className="member__payment"
>
  <span
    className="counter__accept"
  >
    Received
  </span>
  <span
    className="counter__digit digit__margin"
  >
    $
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__digit"
  >
    5
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__digit"
  >
    0
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__digit"
  >
    0
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__digit"
  >
    0
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__digit"
  >
    0
    <span
      className="counter__gradient-top"
    />
  </span>
  <span
    className="counter__from"
  >
    from their top 
    10
     contributing industries
  </span>
</span>
`;
